keymanagementrequirements.md

```diff
@@ -0,0 +1,162 @@
+
+#### Scenarios
+1. A Deployer consumes artifacts from a third party Publisher and wants to configure the trusted publishers in a policy. The Deployer is not the same user/organization as Publisher and wants to independently define the trusted publishers.
+2. A Publisher’s repository may contain a mix of artifacts, published by different teams using different keys. The Deployer wants to trust only artifacts signed by specific keys.
+    - E.g. A group may include the MySQL image from docker hub, and the MySQL Helm chart from another group in the same registry/repo. These are signed by different entities, and valid to be placed in the same repository.
+    
+#### Scenarios for Trust Store Updates
+1. A Deployer updates their dependencies and no longer consumes artifacts from particular Publisher
+2. A public root associated with a signing key a Publisher uses to sign artifacts is compromised.
+3. An root key used by a Publisher is close to expiry. A new key is created and rotated before the existing root expires.
+4. An root key used by a Publisher is not stored securely and the private key is lost. A new key pair is created to continue operations.
+5. A cryptographic algorithm is deprecated, or a compliance standard requires a Publisher to upgrade their key strength.
+
+#### Approaches
+1. Storage and distribution of trust policies from registry
+Pros
+- Trust policies need not be stored in another location
+- Allows in band/automated distribution and update of trust policies to consumers.
+Cons
+- Automated update of trust policies can be disruptive to Deployers that consume artifacts. Deployers should be able to control when trust policies are updated.
+- Only allows Publishers to define the trust policy. It works well when Publisher and Deployer are the same user/organization.
+    - This model does not allow Deployers to independently define trust policies. Furthermore, it may be required that Deployer Admins define trusted publishers, and Deployer Operators only specify/configure which artifacts (from repositories) are deployed.
+
+#### Recommendation
+- Trust policies SHOULD NOT be stored in the registry. Multiple Deployers can consume artifacts from the same repository, and may need to define trust policies independently.
+- Trust policies SHOULD be configured, distributed/updated out of band from artifact updates from registry.
+
+
+### Rescinding Signature Validity
+Artifact Publishers need mechanisms to indicate that a signature they generated is still trustworthy. 
+
+#### Scenarios
+- A Publisher vends a new version of an artifact, and wants to indicate the new version as trusted in addition to older versions already published.
```

Comment:
A second scenario here is that they no longer want to certify a particular artifact (they mistakenly signed/discovered a vulnerability/etc).